### PR TITLE
update Vagrant file 

### PR DESCRIPTION
**Pull request by b4ldr**
This pull updates the Vagrant file as well as the gen_facts scripts to make it easier to genrate facts going forward

### CODE REVIEW

File: facts/get_facts.sh [hunk 1]
```diff
@@ -23,65 +24,71 @@ case "${osfamily}" in
...
-    facter --show-legacy -p -j | tee ${output_file}
+    output_file="/vagrant/`facter --version | cut -c1-3`/`facter operatingsystem | tr '[:upper:]' '[:lower:]'`-`facter operatingsystemmajrelease`-`facter hardwaremodel`.facts"
+    mkdir -p `dirname ${output_file}`
+    [ ! -f ${output_file} ] && facter --show-legacy -p -j | tee ${output_file}
   done
   apt-get install -y make gcc libgmp-dev
   ;;
+'FreeBSD')
+  pkg update -y
+  pkg install -y puppet4 rubygem-bundler
```
Comment (xaque208) [inline facts/get_facts.sh]: `sysutils/puppet5` is now in the tree, as is `sysutils/facter`.

**Comment by xaque208** [comment]
Any authoritative figures of this repo care to comment on this?

**Comment by mcanevet** [comment]
@b4ldr why do you replace `$(...)` with backticks?

**Comment by b4ldr** [comment]
@mcanevet because `$(...)` is bash specific and freebsd does not have bash

**Comment by mcanevet** [comment]
There is way too much things in this PR. Could you please at least split it in different explicit commit?

**Comment by b4ldr** [comment]
Really changing two files, the gitignore and adding a simple 6 line script is too big?  how would you like me to split this up?  TBH i am now spending more time discussing this PR then it took me to make the changes

**Comment by mcanevet** [comment]
This PR could be easily split into multiple commits. For example, you could create one commit per change in the VagrantFile : "Remove debian-6-x86_64", "Change debian-8-x86_64 box", "Remove ubuntu-10.04-x86_64".
It would be easier for us to revert juste one of your commit in case we need to.

**Comment by b4ldr** [comment]
sorry i dont have the time to rework this PR have close it

**Pull request closed without merging**